apple/internal/apple_framework_import.bzl

```diff
@@ -472,10 +497,23 @@ def _apple_static_framework_import_impl(ctx):
 
     return providers
 
+_common_attrs = {
+    "xcframework_platform_ids": attr.string_dict(
```

Comment:
I guess I haven't thought through this enough, but is this something that needs to be specified, versus Bazel figuring out itself?

Reply:
Because Bazel cannot read files at this phase https://github.com/bazelbuild/rules_apple/issues/851#issuecomment-663770260

We could move everything to the tool so that the detection would happen in the execution phase, but that would add more complexity and the imported files would be opaque to Bazel.

All other rules here also just defer the `Info.plist` reading to the rules consumers https://github.com/bazelbuild/rules_apple/blob/54e906af5a45ecd6863bc73d60c7a7419d01bea7/doc/common_info.md#variable-substitution

Reply:
Yeah FWIW i did prototype doing all this automatically, and hit some big blockers like @thii mentioned. The internals need the `File` handles and if you do it more dynamically you're left with just the opaque `TreeArtifact` and it breaks most of the functionality. Still have that commit around if anyone wants to see it again--not super clean but you can see what I was working towards. https://github.com/zachgray-slack/rules_apple/commit/201c0577eaab79e65347164bcb8d1f571d8d3115